src/metabase/driver/postgres.clj

```diff
@@ -95,13 +95,17 @@
        :placeholder "prepareThreshold=0")]))
 
 (defn- enum-types [driver database]
-  (set
-   (map (comp keyword :typname)
-        (jdbc/query (sql-jdbc.conn/connection-details->spec driver (:details database))
-                    [(str "SELECT DISTINCT t.typname "
-                          "FROM pg_enum e "
-                          "LEFT JOIN pg_type t "
-                          "  ON t.oid = e.enumtypid")]))))
+  (try
```

Comment:
In what cases will this fail?

Reply:
It has failed in the past by not acquiring a connection correctly _but_ the rest of the table syncing worked correctly. But checking for this set of enums would take down the entire table checking. I'm happy to back this try/catch out if that's preferable.

Reply:
It has failed in the past by not acquiring a connection correctly _but_ the rest of the table syncing worked correctly. But checking for this set of enums would take down the entire table checking. Happy to revert though if that's preferable.

Reply:
I don't have a strong opinion on this, just trying to understand why (and potentially see if there are any further underlying issues). Perhaps add a comment for posterity.

Reply:
comment was in commit:

> this was throwing errors when we allowed our pooled connections to
close underneath us. But this particular step just isn't important
enough for that. If we don't have a connection available to us, let
the more robust query stuff fail it rather than this one off

can add explicit source comment as well.

Reply:
I've reverted the change so now its just using the correct connection. For two reasons: my code diff was large and it was complaining too much was untested and because phrasing the comment was getting long